sdk/core/System.ClientModel/src/Pipeline/HttpClientPipelineTransport.cs

```diff
@@ -0,0 +1,212 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ClientModel.Primitives;
+
+public partial class HttpClientPipelineTransport : PipelineTransport, IDisposable
+{
+    /// <summary>
+    /// A shared instance of <see cref="HttpClientPipelineTransport"/> with default parameters.
+    /// </summary>
+    public static readonly HttpClientPipelineTransport Shared = new();
+
+    private readonly bool _ownsClient;
+    private readonly HttpClient _httpClient;
+
+    private bool _disposed;
+
+    public HttpClientPipelineTransport() : this(CreateDefaultClient())
```

Comment:
Pri 0: this creates a new HttpClient every time. I thought we would use a shared client in the default case?

Reply:
I think it's ok because we keep a shared instance of the transport, so we only call the constructor once?  But I'll double check - I had thought I had maintained functional parity with the Azure.Core type, but if I didn't I'll fix it.

Reply:
Ok, @KrzysztofCwalina, here is my understanding:
- In Azure.Core, we use a similar pattern for [the HttpClientTransport constructor](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs#L34)
- This doesn't create a new HttpClient every time because we're holding a public static instance of HttpClientTransport in [the transport's Shared property](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs#L65) ... i.e. there is a shared HttpClient instance in the shared transport instance
- System.ClientModel mirrors this pattern with the constructor you've flagged and the corresponding [pubic Shared instance](https://github.com/Azure/azure-sdk-for-net/blob/7de8fbc67a00dcf68e6b8a4169c6061cd651d518/sdk/core/System.ClientModel/src/Pipeline/HttpClientPipelineTransport.cs#L17) we added after yesterday's arch board discussion

If that doesn't address the issue for you, could you help me understand your concern a bit more?  Specifically, what is the use case where we would create multiple instances of HttpClient?

Reply:
Discussed offline with @KrzysztofCwalina.  We can keep a shared instance of HttClient to use from this constructor, but it means we need to do ref-counting to correctly dispose HttpClient.  @KrzysztofCwalina, do you think this should block this PR, or are you ok if I file a release-blocking issue so we can keep moving ClientModel code into main before revisiting?  There may be more P0 issues lurking  that we haven't uncovered yet -- @pallavit has requested we go breadth-first so we can get a good sense of our backlog to set expectations wrt release planning.

Reply:
Filed https://github.com/Azure/azure-sdk-for-net/issues/41301 to track and marked as P0.